src/client/opamCommands.ml

```diff
@@ -3061,6 +3061,75 @@ let clean =
         $logs $switch $all_switches),
   term_info "clean" ~doc ~man
 
+(* LOCK *)
+let lock_doc = "Create locked opam files to share build environments across hosts."
+let lock =
+  let doc = lock_doc in
+  let man = [
+    `S "DESCRIPTION";
+    `P "Generates a lock file of a package: checks the current \
+        state of their installed dependencies, and outputs modified versions of \
+        the opam file with a $(i,.locked) suffix, where all the (transitive) \
+        dependencies and pinnings have been bound strictly to the currently \
+        installed version.";
+    `P "By using these locked opam files, it is then possible to recover the \
+        precise build environment that was setup when they were generated.";
+    `P "If paths (filename or directory) are given, those opam files are locked. \
+        If package is given, installed one is locked, otherwise its latest version.";
+    `P "Fails if all mandatory dependencies are not installed in the switch.";
+    `S "How is locked the opam file?";
+    `P "- $(i,depends) with no filter other than version constraint";
```

Comment:
"depends are fixed to their specific versions, with all filters removed (except for the exceptions below)"